Kernel/Syscalls/mmap.cpp

```diff
@@ -489,9 +490,42 @@ KResultOr<int> Process::sys$munmap(Userspace<void*> addr, size_t size)
         return 0;
     }
 
-    // FIXME: We should also support munmap() across multiple regions. (#175)
+    // Try again while checkin multiple regions at a time
+    // slow: without caching
+    const auto& regions = space().find_regions_intersecting(range_to_unmap);
+    // if there still no regions found error out
+    if (!regions.size())
+        return EINVAL;
 
-    return EINVAL;
+    // check if any of the regions is invalid in somesort, to not accientally
+    // crash with just half a region map left
+    for (auto* region : regions)
+        if (!region->is_mmap())
+            return EPERM;
```

Comment:
Please wrap the outer `for` block in `{}`

Reply:
Done.